agent/structs/config_entry_discoverychain.go

```diff
@@ -1463,3 +1471,39 @@ func IsProtocolHTTPLike(protocol string) bool {
 		return false
 	}
 }
+
+// HTTPHeaderModifiers is a set of rules for HTTP header modification that
+// should be performed by proxies as the request passes through them. It can
+// operate on either request or response headers depending on the context in
+// which it is used.
+type HTTPHeaderModifiers struct {
+	// Add is a set of name -> value pairs that should be appended to the request
+	// or response (i.e. allowing duplicates if the same header already exists).
+	Add map[string]string `json:",omitempty"`
```

Comment:
Is there a use-case when we want to append multiple values to a single key:
```
Foo: bar, baz
```
It looks like a map wouldn't work if trying to register multiple appends to a single key?